lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -731,15 +734,28 @@ def do_3d_projection(self):
         if len(self._linewidths3d) > 1:
             self._linewidths = self._linewidths3d[z_markers_idx]
 
+        PathCollection.set_offsets(self, np.column_stack((vxs, vys)))
+
         # Re-order items
         vzs = vzs[z_markers_idx]
         vxs = vxs[z_markers_idx]
         vys = vys[z_markers_idx]
 
-        PathCollection.set_offsets(self, np.column_stack((vxs, vys)))
+        # Store ordered offset for drawing purpose
+        self._offset_zordered = np.column_stack((vxs, vys))
 
         return np.min(vzs) if vzs.size else np.nan
 
+    @contextmanager
+    def _use_zordered_offset(self):
+        old_offset = self._offsets
+        super().set_offsets(self._offset_zordered)
```

Comment:
Why use `set_offsets` here, but directly setting `self._offsets` below?

Reply:
> Why use `set_offsets` here, but directly setting `self._offsets` below?

Here I see set_offsets does a sanity check before apply new offset to the _offset attribute. Since self._offset_zordered is fresh generated, I have it go through the set_offsets methods. On the other hand old_offset is an attribute directly backup from self._offsets, so direct assignment might be a better choice.